PR: https://github.com/odoo/odoo/pull/85199

From: 66b691dc32d2961e39ec92b45f33619308075a5c
From: Julien Castiaux
Date: 2022-02-23 12:59:52

Structural Changes: 6
Total Changes: 12

Revert "[FIX] website: remove force prefetch for translate fields"

This reverts commit a1a30f52a3162cf3610cbe0885cd85fea3c5e540.

closes odoo/odoo#85199

Related: odoo/enterprise#24660
Signed-off-by: Julien Castiaux <juc@odoo.com>

================================= pseudo patch: =================================

--- a/addons/website/models/ir_ui_view.py
+++ b/addons/website/models/ir_ui_view.py
@@ -389,7 +389,7 @@ class View(models.Model):
                     (request.website.is_public_user() or self.id not in request.session.get('views_unlock', [])):
                 pwd = request.params.get('visibility_password')
                 if pwd and self.env.user._crypt_context().verify(
-                        pwd, self.visibility_password):
+                        pwd, self.sudo().visibility_password):
                     request.session.setdefault('views_unlock', list()).append(self.id)
                 else:
                     error = werkzeug.exceptions.Forbidden('website_visibility_password_required')
@@ -469,10 +469,6 @@ class View(models.Model):
                 editable=editable,
             ))
 
-            # fetch non-prefetchable fields of the mixin SeoMetadata, as they
-            # will be read later in non-sudo mode
-            self.sudo().read(['website_meta_title', 'website_meta_description', 'website_meta_keywords'])
-
         return qcontext
 
     @api.model

--- a/addons/website/models/mixins.py
+++ b/addons/website/models/mixins.py
@@ -21,9 +21,9 @@ class SeoMetadata(models.AbstractModel):
     _description = 'SEO metadata'
 
     is_seo_optimized = fields.Boolean("SEO optimized", compute='_compute_is_seo_optimized')
-    website_meta_title = fields.Char("Website meta title", translate=True)
-    website_meta_description = fields.Text("Website meta description", translate=True)
-    website_meta_keywords = fields.Char("Website meta keywords", translate=True)
+    website_meta_title = fields.Char("Website meta title", translate=True, prefetch=True)
+    website_meta_description = fields.Text("Website meta description", translate=True, prefetch=True)
+    website_meta_keywords = fields.Char("Website meta keywords", translate=True, prefetch=True)
     website_meta_og_img = fields.Char("Website opengraph image")
     seo_name = fields.Char("Seo name", translate=True, prefetch=True)
 
